roles/openshift_health_checker/openshift_checks/diskspace_availability.py

```diff
@@ -0,0 +1,115 @@
+        # is known to support. The only supported driver is currently "devicemapper"
+        if info.get("Driver", "") == self.supported_docker_dev_driver:
+            disktotal = self.to_bytes(status.get("Data Space Total", ""))
+            diskavail = self.to_bytes(status.get("Data Space Available", ""))
+
+            if not disktotal or not diskavail:
+                msg = "Unable to retrieve storage driver information for {driver}.".format(driver=self.supported_docker_dev_driver)
+                return True, msg
+
+            diskleft_percent = diskavail / disktotal * 100
+
+            # fail if available diskspace is less than minimum docker threshold
+            if diskleft_percent < self.minimum_docker_space_percent:
+                msg = "The amount of data space remaining ({remaining}%%) for the docker storage driver is below minimum%%".format(remaining=diskleft_percent, minimum=self.minimum_docker_space_percent)
+                return True, msg
+
+    def noncontainerized_volume_check(self, ansible_mounts, task_vars):
+        recommended_diskspace_gb = self.get_recommended_diskspace_gb(task_vars)
+        diskfree_openshift = self.get_openshift_disk_availability(ansible_mounts)
+
+        if self.to_gigabytes(diskfree_openshift) < recommended_diskspace_gb:
+            msg = "Available disk space ({diskfree} GB) for OpenShift volume below recommended storage. Minimum required disk space: {recommended} GB".format(diskfree=self.to_gigabytes(diskfree_openshift), recommended=recommended_diskspace_gb)
+            return True, msg
+
+        return False, ""
+
+    def get_recommended_diskspace_gb(self, task_vars):
+        group_names = get_var(task_vars, "group_names", default=[])
+        if "masters" in group_names:
+            return self.recommended_master_space_gb
+
+        return self.recommended_node_space_gb
+
+    @staticmethod
+    def get_openshift_disk_availability(ansible_mounts):
+        """Iterates through a map of mounted volumes to determine space remaining on the OpenShift volume"""
+        if not ansible_mounts:
+            return None
+
+        # priority list in descending order
+        supported_mnt_paths = ["/var", "/"]
+        available_mnts = {mnt.get("mount"): mnt for mnt in ansible_mounts}
+
+        for path in supported_mnt_paths:
+            if path in available_mnts:
+                return available_mnts[path].get("size_available")
+
+        return None
+
+    @staticmethod
+    def to_gigabytes(total_bytes):
+        return total_bytes / 1000000000
+
+    @staticmethod
+    def to_bytes(strfmt):
+        units = {
+            "b": 1,
+            "kib": 1024,
+            "kb": 1000,
+            "mib": 1024 ** 2,
+            "mb": 1000 ** 2,
+            "gib": 1024 ** 3,
+            "gb": 1000 ** 3,
+            "tib": 1024 ** 4,
+            "tb": 1000 ** 4,
+            "pib": 1024 ** 5,
+            "pb": 1000 ** 5
```

Comment:
I was not sure what specific units to expect, so this at least covers standard cases. @tbielawa has written [a great python module] that we could use in cases such as this one if we are willing to include it in our openshift-ansible image / make it a requirement on the host machine. wdyt?